pai-management/k8sPaiLibrary/maintainlib/common.py

```diff
@@ -56,6 +56,23 @@ def execute_shell(shell_cmd, error_msg):
 
 
 
+def execute_shell_retry(shell_cmd, clean_cmd, error_msg):
+    count = 0
+    success_flag = False
+    while count < 5 and not success_flag:
+        try:
+            subprocess.check_call( shell_cmd, shell=True )
+            success_flag = True
+        except subprocess.CalledProcessError:
+            count += 1
+            logger.error(error_msg)
+            logger.info("%s error, retrying %d", shell_cmd, count)
+            logger.info("Cleaning failed Pods...")
+            execute_shell_return(clean_cmd, "Can't fine the Pod, cleaning needless...")
+            if count == 5:
```

Comment:
👍 Very readable!
One hint only, `while count < 5` and `if count == 5:` may be duplicated. Can remove one of it.

Reply:
I think no duplicates, it means when it failed 5 times, it will abort the deployment process.

Reply:
I mean, you could `break if count == 5`, or let the `while` loop quit because `count < 5 == false`.